nautobot/utilities/drf_views.py

```diff
@@ -31,249 +27,296 @@
+                # Update custom fields
+                for name in custom_fields:
+                    if name in form.nullable_fields and name in nullified_fields:
+                        obj.cf[name] = None
+                    elif form.cleaned_data.get(name) not in (None, ""):
+                        obj.cf[name] = form.cleaned_data[name]
+
+                obj.full_clean()
+                obj.save()
+                updated_objects.append(obj)
+                self.logger.debug(f"Saved {obj} (PK: {obj.pk})")
+
+                # Add/remove tags
+                if form.cleaned_data.get("add_tags", None):
+                    obj.tags.add(*form.cleaned_data["add_tags"])
+                if form.cleaned_data.get("remove_tags", None):
+                    obj.tags.remove(*form.cleaned_data["remove_tags"])
+
+            # Enforce object-level permissions
+            if self.queryset.filter(pk__in=[obj.pk for obj in updated_objects]).count() != len(updated_objects):
+                raise ObjectDoesNotExist
+        if updated_objects:
+            msg = f"Updated {len(updated_objects)} {model._meta.verbose_name_plural}"
+            self.logger.info(msg)
+            messages.success(self.request, msg)
+        self.success_url = self.get_return_url(request)
+
+    def _bulk_create(self, form):
+        """
+        Helper method to create objects in bulk after the form is validated successfully.
+        """
+        # Iterate through CSV data and bind each row to a new model form instance.
+        new_objs = []
+        request = self.request
+        with transaction.atomic():
+            headers, records = form.cleaned_data["csv_data"]
+            for row, data in enumerate(records, start=1):
+                obj_form = self.import_form(data, headers=headers)
+                restrict_form_fields(obj_form, request.user)
+
+                if obj_form.is_valid():
+                    obj = self._save_obj_for_bulk_import(obj_form, request)
+                    new_objs.append(obj)
+                else:
+                    for field, err in obj_form.errors.items():
+                        form.add_error("csv_data", f"Row {row} {field}: {err[0]}")
+                    raise ValidationError("")
+
+            # Enforce object-level permissions
+            if self.queryset.filter(pk__in=[obj.pk for obj in new_objs]).count() != len(new_objs):
+                raise ObjectDoesNotExist
+
+        # Compile a table containing the imported objects
+        self.obj_table = self.table_class(new_objs)
+        if new_objs:
+            msg = f"Imported {len(new_objs)} {new_objs[0]._meta.verbose_name_plural}"
+            self.logger.info(msg)
+            messages.success(request, msg)
+
+    def form_valid(self, form):
+        self.logger.debug("Form validation was successful")
+        request = self.request
+
+        if self.action == "destroy":
+            self._destroy()
```

Comment:
Might be better from a self-consistency standpoint for `_destroy()` to take a `form` parameter, even if it doesn't do anything particular with it at present?